File: src/linkedList/java/palindrome/twoPointerApproach.java
package linkedList.java.palindrome;

import linkedList.java.Node;
import java.util.List;
import java.util.ArrayList;
import java.util.Objects;

public class twoPointerApproach {
	public boolean isPalindrome(Node curr){
		List<Integer> contents = new ArrayList<>();
		while(curr != null){
			contents.add(curr.data);
			curr = curr.next;
		}

		int left = 0, right = contents.size() - 1;
		while(left < right){
			if(!Objects.equals(contents.get(left), contents.get(right))){
				return false;
			}
			left += 1;
			right -= 1;
		}
		return true;
	}
}

File: src/linkedList/python/palindrome/twoPointerApproach.py
from linkedList.python.node import node


def is_palindrome(curr: node) -> bool:
    contents = []
    while curr:
        contents.append(curr.data)
        curr = curr.next

    left = 0
    right = len(contents) - 1
    while left < right:
        if contents[left] != contents[right]:
            return False
        left += 1
        right -= 1
    return True
